server/redirect-url-helper.js

```diff
@@ -33,11 +33,17 @@ function processRedirects(req, res, next, sourceUrlArray, urls) {
           });
         }
         const dynamicKeys = extractedSourceUrl(req.path);
-        const compiledPath =
-          dynamicKeys && extractedDestinationUrl(dynamicKeys.params);
+        const compiledPath = dynamicKeys && extractedDestinationUrl(dynamicKeys.params);
         if (compiledPath) {
+          const validStatusCodes = [301, 302];
+          const statusCode = parseInt(urls[sourceUrl].statusCode, 10);
+          if (validStatusCodes.includes(statusCode)) {
+            const cacheValue = statusCode === 301 ? "max-age=604800" : "max-age=86400";
+            res.set("cache-control", `public,${cacheValue}`);
+          }
```

Comment:
This can be much more optimized, feel free to modify as needed / relevant.

```suggestion
          const validStatusCodes = {301: "max-age=604800", 302: "max-age=86400"};
          const statusCode = parseInt(urls[sourceUrl].statusCode, 10);
          const statusValue = validStatusCodes[statusCode];
          if (statusValue) {
            res.set("cache-control", `public,${statusValue}`);
          }
```

Reply:
There is no perf gain with this as the input size is fixed(301,302), but this seems to be a better pattern. I will make this change

Reply:
Done